### PR TITLE
bazel-watcher: 0.5.0 -> 0.9.0

### PR DESCRIPTION
**Pull request by uri-canva**
###### Motivation for this change

Upgrade `bazel-watcher` so that it can be built with newer versions of Bazel.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/tools/bazel-watcher/default.nix [hunk 1]
```diff
@@ -54,7 +48,7 @@ buildBazelPackage rec {
       sed -e '/^FILE:@go_sdk.*/d' -i $bazelOut/external/\@*.marker
     '';
 
-    sha256 = "1iyjvibvlwg980p7nizr6x5v31dyp4a344f0xn839x393583k59d";
+    sha256 = if stdenv.isDarwin then "1zlg8nay3ij0gnq3ip33cz4f33jhkb1y05n5y8w98hcnkmx3m1vs" else "11pimgaxfx0p003zdxjx9fc9whgxh7qmycihakki4hm2n9ys22fq";
```
Comment (uri-canva) [inline pkgs/development/tools/bazel-watcher/default.nix]: https://github.com/bazelbuild/bazel-gazelle now includes native binaries after adding it to the workspace, so the hashes will be different depending on the architecture.

**Comment by uri-canva** [comment]
cc @kalbasit

**Review by kalbasit — COMMENTED** [review]
I'm on vacation until the 26th with limited internet and no laptop. The code looks great, but I'd love to make sure it runs before approving. /cc @Mic92 if he's available to review.

**Comment by Mic92** [comment]
@GrahamcOfBorg build  bazel-watcher

**Comment by Mic92** [comment]
I got locally:

```
 fixed-output derivation produced path '/nix/store/c2hzc13ajagwfdfl4qiafvl5fq046j0q-bazel-watcher-0.9.0-deps' with sha256 hash '03knj3jmn5nn36kf3m4s30b4ms7hhmd1k9rcdzy70s8q6v2y84yr' instead of the expected hash '11pimgaxfx0p003zdxjx9fc9whgxh7qmycihakki4hm2n9ys22fq'
```

**Comment by kalbasit** [comment]
I got

```
fixed-output derivation produced path '/nix/store/jcnzfn13w02jn3lbf08pzq68jjz5grv7-bazel-watcher-0.9.0-deps' with sha256 hash '0239h6z605npjv8kpw3h3jkc0zlvbpvrr4xm30dvr9n5s9piqlnq' instead of the expected hash '11pimgaxfx0p003zdxjx9fc9whgxh7qmycihakki4hm2n9ys22fq'
```

I think the `buildBazelPackage` needs to be adjusted again. I uploaded /nix/store/jcnzfn13w02jn3lbf08pzq68jjz5grv7-bazel-watcher-0.9.0-deps [here](http://s000.tinyupload.com/index.php?file_id=06863169693901591115), can you compare the files with your version and adjust `buildBazelPackage` to remove those differences? See 86a5535b2fdb6fbb88ca83ba9d83851269241927 for my latest attempt at removing those differences.

**Comment by uri-canva** [comment]
Sorry was on vacation too. Will take a look.

**Comment by uri-canva** [comment]
Seems like the binaries include the output path in them, and `strip -S` doesn't strip it, so it's not a debug symbol. There some information in https://blog.filippo.io/reproducing-go-binaries-byte-by-byte/ about reproducible go binaries, it might be related to the build id.

```
$ go run gosym.go fetch_repo
/tmp/nix-build-bazel-watcher-0.9.0-deps.drv-0/output/external/bazel_gazelle_go_repository_tools/src/github.com/bazelbuild/bazel-gazelle/cmd/fetch_repo/fetch_repo.go
$ go run gosym.go ~/Downloads/nix/store/jcnzfn13w02jn3lbf08pzq68jjz5grv7-bazel-watcher-0.9.0-deps/bazel_gazelle_go_repository_tools/bin/fetch_repo
/build/output/external/bazel_gazelle_go_repository_tools/src/github.com/bazelbuild/bazel-gazelle/cmd/fetch_repo/fetch_repo.go
```

**Comment by Mic92** [comment]
why is this a fixed-input derivation anyway?

**Comment by uri-canva** [comment]
You just gave me an idea though.

**Comment by uri-canva** [comment]
Done, change it to delete the gazelle binary tools since they can be rebuilt during the build, because it doesn't require accessing the network.

**Comment by kalbasit** [comment]
cool, let me give it another try! Thank you @uri-canva for working on this!

**Review by kalbasit — APPROVED** [review]

**Pull request merged**